cpp/src/io/statistics/byte_array_view.cuh

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/utilities/span.hpp>
+
+namespace cudf::io {
+
+class byte_array_view {
+ public:
+  using element_type = int8_t const;  ///< The type of the elements in the byte array
```

Comment:
For parquet, at least, I think this would be better as uint8_t since parquet assumes unsigned byte-wise ordering for binary types

Reply:
Yes, I have a pending change for it to be a uint8_t since @devavret pointed out that `std::byte` was `uint8_t` as well.